apps/web/src/server/actions/organization.ts

```diff
@@ -34,36 +76,311 @@ export async function getOrganizationProjects(id: Organization["id"]) {
   return projects;
 }
 
-export async function getOrganizations() {
+export async function getOrganizations(): Promise<
+  | [
+      {
+        orgId: number | null;
+        orgName: string | null;
+      }[],
+      null,
+    ]
+  | [null, Error]
+> {
   const session = await getServerAuthSession();
 
   if (!session) {
-    throw new Error("You must be signed in to perform this action");
+    return [null, new Error("You must be signed in to perform this action")];
+  }
+
+  return [
+    await db
+      .select({
+        orgId: organizations.id,
+        orgName: organizations.name,
+      })
+      .from(usersToOrganizations)
+      .leftJoin(
+        organizations,
+        eq(usersToOrganizations.organizationId, organizations.id),
+      )
+      .where(eq(usersToOrganizations.memberId, session.user.id)),
+    null,
+  ];
+}
+
+export async function createOrganization({ name }: InsertOrganization) {
+  const session = await getServerAuthSession();
+  try {
+    const newOrg = (
+      await db
+        .insert(organizations)
+        .values({ name })
+        .returning({ id: organizations.id, name: organizations.name })
+        .execute()
+    )[0]!;
+
+    await db
+      .insert(usersToOrganizations)
+      .values({
+        memberId: session!.user.id,
+        organizationId: newOrg.id,
+        role: "owner",
+      })
+      .execute();
+
+    return newOrg;
+  } catch (e) {
+    throw new Error("Organization name already exists");
   }
+}
+
+export async function updateOrganizationName({
+  oldName,
+  name,
+}: {
+  oldName: string;
+  name: string;
```

Comment:
We should get the types from those inherited from our db. So this should be 
```suggestion
  oldName: Organization["name"];
  name: Organization["name"];
```

Reply:
fixed